src/util/endian.rs

```diff
@@ -0,0 +1,103 @@
+#[inline]
+pub fn slice_to_u32_be(slice: &[u8]) -> u32 {
+    assert_eq!(slice.len(), 4);
+    (slice[0] as u32) << 3*8 |
+    (slice[1] as u32) << 2*8 |
+    (slice[2] as u32) << 1*8 |
+    (slice[3] as u32) << 0*8
+}
+
+#[inline]
+pub fn u32_to_array_be(val: u32) -> [u8; 4] {
+    [
+        ((val >> 3*8) & 0xff) as u8,
+        ((val >> 2*8) & 0xff) as u8,
+        ((val >> 1*8) & 0xff) as u8,
+        ((val >> 0*8) & 0xff) as u8,
+    ]
+}
+
+#[inline]
+pub fn u16_to_array_le(val: u16) -> [u8; 2] {
+    [
+        ((val >> 0*8) & 0xff) as u8,
+        ((val >> 1*8) & 0xff) as u8,
+    ]
+}
+#[inline]
```

Comment:
Since we're not using rustfmt here: style comments!

Please newline in between methods.